build/bootstrap.proj

```diff
@@ -0,0 +1,31 @@
+<Project Sdk="Microsoft.Build.NoTargets/1.0.80">
+    <PropertyGroup>
+        <RestorePackagesPath>$(MSBuildThisFileDirectory)../packages/</RestorePackagesPath>
+        <BaseIntermediateOutputPath>$(RestorePackagesPath)obj/</BaseIntermediateOutputPath>
```

Comment:
We should use MSBuildProjectExtensionsPath instead of BaseIntermediateOutputPath. 
Right now I'd imagine it's like that the assets file ends up in some unexpected place.

Reply:
Sounds good but, I would need to import the props and targets of the project. Let me see how it works.

Reply:
> I would need to import the props and targets of the proj

Not sure what you mean by this.

Reply:
I mean adding import statements to load `\build\common.props` and `"$(BuildCommonDirectory)common.targets` from this repo (NuGet.Client)

Reply:
Isn't it the other way around?


Specifically the following should be enough.

```
    <PropertyGroup>
        <RestorePackagesPath>$(MSBuildThisFileDirectory)../packages/</RestorePackagesPath>
        <MSBuildProjectExtensionsPath>$(RestorePackagesPath)obj/</MSBuildProjectExtensionsPath>
        <TargetFramework>netstandard2.0</TargetFramework>
        <DisableImplicitNuGetFallbackFolder>true</DisableImplicitNuGetFallbackFolder>
    </PropertyGroup>
```

Reply:
Oh. I didn't know that. Let me test this change

Reply:
It works perfectly. I suggest to move the project.assets.json of this bootstrap.proj to `.\artifacts\packages\obj\project.assets.json` to avoid mixing this file with the packages folder

Reply:
I'm ok with that. 
The caches and assets file are only used when configure.ps1 is invoked so the footprint should be negligible.

Reply:
FYI I intentionally did not put the assets file in the `artifacts` directory because some build steps like signing and localisation takes all assemblies in the `artifacts` directory. Now, this bootstrap project doesn't create assemblies, so it's potentially not a problem, but in general test projects don't put their assemblies in `artifacts` either. 

Since the `artifacts` folder appears to be mostly for projects that get shipped to customers, it didn't feel appropriate to me to put the bootstrap assets file there.